game_for_class.py

```diff
@@ -8,70 +8,46 @@
 # Start with the player's class
 class Character:
     """All the methods the game player can use"""
-    def __init__(self,
-                 name: str,
-                 hp: float,
-                 attack: float,
-                 defense: float,
-                 speed: float,
-                 team: int):
+    def __init__(self, name: str, hp: float, attack: float, defense: float, speed: float, team: int):
+
         self.name = name
         self.hp = hp
         self.attack = attack
         self.defense = defense
         self.speed = speed
         self.team = team
+        """Initializes the Character"""
 
     def is_alive(self) -> bool:
-        """Returns true if the character is alive"""
+
+        """returns true if the player is alive"""
         return self.hp > 0
 
-    def is_enemy(self, other_player) -> bool:
-        """Returns true if the player is an enemy"""
-        return other_player.team != self.team
+    def is_ally(self, other_player) -> bool:
+        """returns true if the other player is an ally"""
+        return other_player.team == self.team
 
-    def deal_damage(self, other_player: 'Character') -> 'Character':
-        """Deals damage to another Charater Object
+    def get_all_enemy_indices(self, players: List['Character']) -> List[int]:
+        """returns a list of enemies to chose from"""
+        return [index for index, player in enumerate(players) if not self.is_ally(player) and player.is_alive()]
 
-        :param other_player: TODO: Fill in
-        :return: TODO: Fill in
-        """
-        if other_player.is_alive:
+    def deal_damage(self, other_player):
+        """deals damage to an enemy if they are alive"""
+        if other_player.is_alive():
             other_player.hp -= self.attack
-        # Alex: Moved this out of the if statement
-        # because this method wouldn't have returned a value if the other player
-        # was dead.
         return other_player
 
-    def get_all_enemy_indices(self,
-                              players: List['Character']) -> List[int]:
-        """Returns a list of enemies to choose a target from
-
-        Specifically, returns the indices in players that are characters
-        which are both enemies and alive
-
-        :param players: TODO: Fill in
-        :return: TODO: Fill in
-        """
-        return [
-            index for index, player in players
-            if player.is_alive and player.is_enemy
-        ]
+    def block(self, other_player) -> bool:
+        """Blocks for a little bit of damage before returning to normal damage"""
+        if self.block():
+            return self.block == True
+        if self.block == True:
+            other_player.attack -= self.defense
```

Comment:
This will permanently lower the opponents damage.

Think about this. How can you handle blocking during `deal_damage`? `block` should just set the player's mode to block (you will want to add a attribute to the `Character` called `self.blocking` and just have the `block` method set `self.blocking` to `true`.

During `deal_damage`, you can just handle the blocking there.

This is a lot of hints on how to do this. Let me know if you have questions, but I don't want to hold your hand on this too much because it won't be your project anymore.

Reply:
Okay, so if I try having an if statement that means if self.blocking == true, lower the attack, but else, (if self.blocking == false, then have the program set the attack to normal regardless of lowered eg:
    def deal_damage(self, other_player):
        if other_player.is_alive():
            other_player.hp -= self.attack
            if other_player.blocking == True:
                self.attack -= other_player.defenss
            else:
                
            return other_player

am I going in the right direction?